CONTRIBUTING.md

```diff
@@ -151,30 +136,72 @@ Once you've run this, your repository should be set for most changes (i.e., you
 
 ### 4. Make the changes you've discussed
 
-Try to keep the changes focused to the issue. We've found that working on a [new branch][link_branches] for each issue makes it easier to keep your changes targeted. 
+Try to keep the changes focused to the issue.  
+We've found that working on a [new branch][link_branches] for each issue makes it easier to keep your changes targeted.
+Using a new branch allows you to follow the standard "fork/branch/commit/pull-request/merge" GitHub 
+workflow when making changes.
+[This guide][link_gitworkflow] provides a useful overview for this workflow.
+Before making a new branch, make sure your master is up to date.
 
-Using a new branch allows you to follow the standard "fork/branch/commit/pull-request/merge" GitHub workflow when making changes. [This guide][link_gitworkflow] provides a useful overview for this workflow.
+```
+git checkout master
+git fetch upstream master
+git merge upstream/master
+```
+
+Then, make your new branch.
+
+```
+git checkout -b MYBRANCH
+```
+
+As you're making changes, make sure your branch is kept up to date with 
+
+```
+git fetch upstream master
+git merge upstream/master
+```
+
+If you know what rebasing is, please only use it for changes that haven't been pushed.
```

Comment:
The problem I've run into is that new git users will look up how to do something, and then see a StackExchange post saying to rebase or some forum post saying to do so, and then they do so without realizing the consequences. For example, the Linux kernel frequently recommends a rebase, but they're also targeting users much more git-familiar than a typical contributor in open source science projects. This has happened a few times, so I would like to do something to indicate to people that we don't recommend it unless they're sure of what they're doing.

Reply:
Maybe we could say something more generally about not amending history in pull requests ? So it would cover git commit --amend, too ? We can link here: https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

And point out that rewriting history can make it more difficult for others to work collaboratively with you on a new feature or bug fix ?

Reply:
That makes sense. I always forget that you can `--amend` because usually I do my cleanup as an interactive rebase. I probably should have dropped the link in there to begin with, it's the link I use in my tutorials for my lab. I'll go ahead and work it into that section, thanks!

Reply:
Do you like how it's handled in 1bd9b19?

Reply:
I've opted to just delete this part since we like how it's handled in the other section. I figure it's somewhat unlikely for someone to attempt a rebase at that point unless they're aware of that version of the workflow.